rtrlib/rtr_mgr.c

```diff
@@ -292,16 +293,19 @@ int rtr_mgr_config_cmp_tommy(const void *a, const void *b)
 	return rtr_mgr_config_cmp(ar->group, br->group);
 }
 
+// TODO: Additional arguments trailing?
```

Comment:
Is the TODO comment still needed? If yes, I suggest to clarify it, I don't understand it.

Reply:
Due to the added aspa update fp, we’re introducing a source-breaking change here.. The alternative would’ve been to add the aspa fp argument as the last parameter (at the expense of consistency, still source-breaking)

Reply:
We thought about it a lot how we could avoid breaking user space, but there was always at least one bitter pill to swallow. Question is now which bitter pill do you want @fho.

Reply:
Difficult, it's not even clear to me how bad it would be to break compatibility.
rtrlib  still has major version 0, which allows downwards incompatible changes according to the [semver](https://semver.org/) spec.
It would be good to release version 1 some when and guarantee API compatibility. 

The following alternatives come to my mind, you probably also discussed them already:
- keep `rtr_mgr_init` as it is, introduce a `rtr_mgr_init1` or `rtr_mgr_init_with_aspa` function that does the same but also has the new apsa_update_fp parameter
- keep `rtr_mgr_init` as it is, add a new function that only sets  the `aspa_update_fp` (`rtr_mgr_register_aspa_callback()` or so).
For version 1 rtr_mgr_init could also be refactored to  set all update_fps via separate functions.

Reply:
@carl-tud @tanneberger any thought regarding https://github.com/rtrlib/rtrlib/pull/285#discussion_r1526483840?

Reply:
good reminder - thank you, I actually built this a couple of weeks ago and just forgot to push it.

I added the `aspa_array_reserve` function which can pre-allocate memory and started using it at all places, where we know the size already before.

Reply:
@fho I personally would just change the signature of `rtr_mgr_init` and then create a new release (v0.9.0). 

- This gives people the option to sick with the current release and if there is anything critical, we update both versions until everybody migrated.

Reply:
> @fho I personally would just change the signature of rtr_mgr_init and then create a new release (v0.9.0).

Then lets go with that! :-)

>Other idea is to do a proper refactor: - What is your opinion on this? :

I like that it introduces an API that could make it easier in the future to add more functionality while keeping the API downwards compatible. I don't know if more functionality that requires a `rtr_mgr_init` signature change will be added in the future though.
To fully initialize the rtrlib 3 additional function calls would always be needed. I wonder if there are use cases where you don't want to have the full functionality. If there aren't or only very rarely, it makes the initialization more cumbersome

Reply:
But I guess we still can add functions like: `rtr_mgr_init_all(aspa_fp, spki_fp, roa_fp);`